e2e/testcases/helm_sync_test.go

```diff
@@ -136,6 +139,354 @@ func TestPublicHelm(t *testing.T) {
+		TypeMeta:   configMapTypeMeta,
+		ObjectMeta: configMapObjMeta,
+		Data: map[string]string{"values.yaml": `
+image:
+  digest: sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e
+  pullPolicy: Always
+wordpressUsername: test-user-1
+wordpressEmail: override-this@example.com
+resources:
+  requests:
+    cpu: 150m
+    memory: 250Mi
+  limits:
+    cpu: 1
+    memory: 300Mi
+mariadb:
+  primary:
+    persistence:
+      enabled: false
+service:
+  type: ClusterIP`},
+	}
+	cm1Copy := cm1.DeepCopy()
+	if err := nt.KubeClient.Create(&cm1); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	rs := fake.RootSyncObjectV1Beta1(configsync.RootSyncName)
+	nt.MustMergePatch(rs, `{
+		"spec": {
+		  "sourceFormat": "unstructured",
+		  "sourceType": "helm",
+		  "helm": {
+			"releaseName": "my-wordpress",
+			"namespace": "wordpress",
+			"auth": "none",
+			"repo": "https://charts.bitnami.com/bitnami",
+			"chart": "wordpress",
+			"version": "15.2.35",
+			"values": {
+			  "extraEnvVars": [
+				{
+				  "name": "TEST_1",
+				  "value": "val1"
+				},
+				{
+				  "name": "TEST_2",
+				  "value": "val2"
+				}
+			  ],
+			  "wordpressEmail": "test-user@example.com"
+			},
+			"valuesFrom": [
+			  {
+				"kind": "ConfigMap",
+				"name": "foo"
+			  }
+			]
+		  }
+		}
+	  }`)
+
+	err := nt.WatchForAllSyncs(nomostest.WithRootSha1Func(helmChartVersion("15.2.35")),
+		nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{nomostest.DefaultRootRepoNamespacedName: "wordpress"}))
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	var expectedCPURequest string
+	var expectedCPULimit string
+	var expectedMemoryRequest string
+	var expectedMemoryLimit string
+	if nt.IsGKEAutopilot {
+		expectedCPURequest = "250m"
+		expectedCPULimit = expectedCPURequest
+		expectedMemoryRequest = "512Mi"
+		expectedMemoryLimit = expectedMemoryRequest
+	} else {
+		expectedCPURequest = "150m"
+		expectedCPULimit = "1"
+		expectedMemoryRequest = "250Mi"
+		expectedMemoryLimit = "300Mi"
+	}
+	if err := nt.Validate("my-wordpress", "wordpress", &appsv1.Deployment{}, containerImagePullPolicy("Always"),
+		testpredicates.HasCorrectResourceRequestsLimits("wordpress",
+			resource.MustParse(expectedCPURequest),
+			resource.MustParse(expectedCPULimit),
+			resource.MustParse(expectedMemoryRequest),
+			resource.MustParse(expectedMemoryLimit)),
+		testpredicates.HasExactlyImage("wordpress", "bitnami/wordpress", "", "sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "WORDPRESS_USERNAME", "test-user-1"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "WORDPRESS_EMAIL", "test-user@example.com"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "TEST_1", "val1"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "TEST_2", "val2")); err != nil {
+		nt.T.Error(err)
+	}
+	if nt.T.Failed() {
+		nt.T.FailNow()
+	}
+
+	nt.T.Log("Apply the ConfigMap with values to the cluster")
```

Comment:
Can you update the log string here (and elsewhere) to be a bit more descriptive? From looking at the diff in the values file I assume the important bit in this case is updating pullPolicy from `always` -> `never`. Can save the reader some time by making the message more descriptive

Reply:
done